### ISSUE TITLE
[Feature Request]: To have a status bar which can show parallel execution between the steps.

### ISSUE THREAD

**Issue by jodtribhu** [carbon-design-system/carbon#13960]
### The problem

In our React UI , we have a requirement to show parallel stages of the status bar like the execution of step 1 and step 3 in progress even when the step 2 is completed and to show description for it too.But this cannot be done with the current progress bar in the Carbon Design system as it is indicated like Step 1, Step 2 which will be like executing the stages one after the another instead of parallel execution

### The solution
Why not provide a status bar which can take inputs of three types InProgress, Completed and Failed which also has a hover text to show description of each of the stages? I have worked on a sample to show a status bar which are showing parallel execution.

### Examples

https://github.com/carbon-design-system/carbon/assets/51241236/90faccb6-d790-40bf-9ecb-435ba46f4869

https://github.com/carbon-design-system/carbon/assets/51241236/2d1c8682-0fc4-434f-bd51-94cd7ebd04c0

### Business priority

Medium Priority = upcoming release but is not pressing

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)

**Comment by sstrubberg** [comment]
Like this?

https://react.carbondesignsystem.com/?path=/story/components-progressindicator--default

**Comment by jodtribhu** [comment]
Yes I had seen this @sstrubberg but why we tried out a new design is that because that we want to have to show the steps like step 2 and step 4 in inProgress while the step 3  and step 1 is completed.This scenario comes because we are executing the step1 and step3 in parallel and once it has been completed it will move on to the next step.

**Comment by sstrubberg** [comment]
I'd like to see more use-cases where this is used in Product before we take this on. What product are you supporting?

**Comment by jodtribhu** [comment]
Hi @sstrubberg sorry for the late reply some of the use cases which this can be used is in the case of pipeline where the execution of tasks can be done in parallel.

**Comment by jodtribhu** [comment]
Hi @sstrubberg is there any update on this feature ?

**Comment by jodtribhu** [comment]
There are many use case which it can be used where we require parallel execution or linear flow..
